### DISCUSSION TITLE
netavark architecture diagram?

### DISCUSSION THREAD

**Discussion by iKloudly** [containers/netavark#1006] [Q&A] [answered]
Is there some type of an architecture diagram for `netavark` by any chance? Just curious how it works under the hood!

**Comment by Luap99** [comment, marked as answer]
What kind of architecture, how it works internally and communicates with podman or how it configures the networking?

For the later I gave a talk at devconf.cz last week (https://pretalx.com/devconf-cz-2024/talk/BVM77L/), you can find the slides there which includes a diagram for rootful and rootless bridge setup.

**Reply by iKloudly** [reply]
Ooh, this is AWESOME! thank you so much.
I was originally watching https://www.youtube.com/watch?v=tlxDmUPc4WY&t=252s and I was hoping there's more to it.. I really appreciate the resource, thank you!
